accessor-alias/sycl-2.2/index.md

```diff
@@ -0,0 +1,686 @@
+  * [Defaulting template parameters](#defaulting-template-parameters)
+  * [Simplifying access modes](#simplifying-access-modes)
+  * [Aliases](#aliases)
+  * [Extending the handler](#extending-the-handler)
+* [Considerations and alternatives](#considerations-and-alternatives)
+
+### TLDR: [Examples](#examples)
+
+## Motivation
+
+Accessors are the cornerstone of SYCL
+as they are used inside and outside kernels to access data
+in a pointer-like manner.
+However, SYCL ecosystem projects have shown that
+they can be a bit awkward to handle due to their verbosity.
+Storing accessors in function objects requires specifying all template parameters,
+and accessors of different access modes are treated as different types,
+even though there isn't much difference in accessing data.
+
+### Existing `local_accessor` alias
+
+SYCL 1.2.1 already has an alias for local memory,
+which can serve as an illustrative example of an alias reducing verbosity.
+
+```cpp
+// Without the alias
+accessor<int,
+          1,
+          access::mode::read_write,
+          access::target::local>
+    localAccOld{range<1>{32}};
+
+// With the alias
+local_accessor<int>
+    localAcc{range<1>{32}};
+```
+
+## Revisions
+
+### 0.2
+
+* Simplified handling of non-writeable accessors
+* Defined implicit conversions w.r.t. constness and access modes
+* Removed `buffer_accessor` alias
+* Removed type traits
+* Removed member functions for returning aliases
+* Added `access::mode` template parameter to `host_accessor`
+* Removed section _Discouraging access mode combinations_
+* Discuss CTAD
+* Discuss trait for writeable accessors
+* More and better examples
+* Added table of contents
+* Deprecate `access::mode::atomic` as part of this proposal
+* Link to proposal that deprecates `accessor::placeholder`
+* Minor fixes
+
+### 0.1
+
+* Initial proposal
+
+## Summary
+
+Main points of the proposal:
+
+1. [Default `accessor` template parameters](#defaulting-template-parameters)
+1. [Clarify and simplify access modes](#simplifying-access-modes)
+    * `read` and `read_write` most important
+    * Allow `const T` to denote read-only data access
+    * Deprecate `access::mode::atomic`
+1. [Introduce `access::target`-specific aliases for accessors](#aliases)
+1. [Extending `handler::require`](#extending-the-handler)
+
+The proposal slightly changes the semantics of accessing data,
+but still aims to be completely backwards compatible with SYCL 1.2.1.
+
+Note that this proposal assumes that any accessor can be a placeholder,
+essentially deprecating the `access::placeholder` template parameter.
+This is covered by a separate proposal:
+[CP024 Default placeholders](https://github.com/codeplaysoftware/standards-proposals/pull/122).
+It also doesn't address image accessors.
+We discuss this and some other issues
+in the [Considerations and alternatives](#considerations-and-alternatives) section.
+
+## Defaulting template parameters
+
+We propose defaulting all accessor template parameters
```

Comment:
This could conflict with the changes proposed in https://github.com/KhronosGroup/SYCL-Docs/pull/73 to introduce an optional `property_list` to thee `accessor` constructor. Though perhaps some of the information that we currently pass such as the `access::mode` information that is not needed for the type-system could be passed as a property?

Reply:
I don't quite see how that conflicts, this proposal is about defaulting template parameters, not constructor arguments. Though it's possible some extra deduction guides might be needed because of those changes.